internal/host/validator.go

```diff
@@ -1626,3 +1635,73 @@ func (v *validator) noSkipMissingDisk(c *validationContext) (ValidationStatus, s
 	}
 	return ValidationSuccess, successMessage
 }
+
+func (v *validator) serviceCanConnectToSpokeKubeAPI(c *validationContext) (ValidationStatus, string) {
+
+	// This only applies to scenarios where we have a cluster, kube-api is enabled and the cluster is in "Day2" mode.
+	if c.cluster == nil || !c.kubeApiEnabled || *c.cluster.Status != models.ClusterStatusAddingHosts {
+		return ValidationSuccessSuppressOutput, "Host is not part of a Day2 cluster, no need to check KubeAPI connectivity"
+	}
+
+	// Download the spoke cluster config from internal storage.
+	respBody, contentLength, err := c.objectHandler.Download(c.ctx, fmt.Sprintf("%s/%s", c.host.ClusterID, constants.Kubeconfig))
+	if err != nil {
+		return ValidationError, "Could not download spoke cluster kubeconfig from internal storage"
+	}
+
+	// Define a buffer to hold the kubeconfig
+	buffer := make([]byte, contentLength)
+
+	// Calling ReadAtLeast method with its parameters
+	bytesRead, err := io.ReadAtLeast(respBody, buffer, int(contentLength))
+
+	// If error is not nil then panics
+	if err != nil {
+		return ValidationError, fmt.Sprintf("Could not read spoke cluster kubeconfig from internal storage: %s", err.Error())
+	}
+	if bytesRead > int(contentLength) {
+		return ValidationError, "Too many bytes read when reading spoke cluster kubeconfig from internal storage"
+	}
+
+	// Using the Kubeconfig, obtain a client
+	spokeK8sClientFactory := common.NewSpokeK8sClientFactory(v.log)
+	client, err := spokeK8sClientFactory.CreateFromKubeconfig(buffer)
+	if err != nil {
+		return ValidationFailure, "Could not create the spoke k8s client connection using kubeconfig"
+	}
+
+	namespace := c.cluster.KubeKeyNamespace
+	csrListSAR := &authorizationv1.SelfSubjectAccessReview{
+		Spec: authorizationv1.SelfSubjectAccessReviewSpec{
+            ResourceAttributes: &authorizationv1.ResourceAttributes{
+                Namespace: namespace,
+                Verb:      "list",
+                Resource:  "CertificateSigningRequest",
+            },
+        },
+	}
+	sarResponse, err := client.CreateSubjectAccessReview(csrListSAR)
+	if  err != nil {
+		return ValidationFailure, "Could not create subject access review to detemine csr list permissions"
+	}
+	if !sarResponse.Status.Allowed {
+		return ValidationFailure, "The spoke does not permit the listing of CSRs"
```

Comment:
Remember this is a user-facing message. I wouldn't say the "the spoke doesn't permit the listing of CSRs" as that is not true in the general sense. I think the message should be more specific - "The cluster credentials available to the Assisted Installer do not have permissions to <verb> <resource>s on the spoke cluster"